### DISCUSSION TITLE
Async operation in gatsby-config.js

### DISCUSSION THREAD

**Discussion by lucashfreitas** [gatsbyjs/gatsby#32273] [Help] [answered]
The configuration needed by one of the plugins lives in a external source, like an API/database. I would like to fetch them before using in `gatsby-config.js`. Would be possible to call my async function in `gatsby-config.js` or in the plugin options support async operations?

**Comment by KyleAMathews** [comment, marked as answer]
If you need to fetch config for multiple plugins, we suggest you run a script before starting gatsby which downloads the config and writes it to a file that gatsby-config.js can require.

If it's for a specific plugin, it can fetch the config options itself when it runs as all hooks support running async.

**Reply by lucashfreitas** [reply]
Hello @KyleAMathews , I ended up running a script that writes all data to a JSON file which is required later by the gatsby-config.js file. Unfortunately the plugin `gatsby-plugin-feed` does not offer a way to have multiple feeds created from graphql (we could use onPostBuild to do that but would require modification in the source code).  

https://github.com/gatsbyjs/gatsby/issues/12184 For anyone running into the same issue. 

@KyleAMathews all the plugins under gatsby main repo package directory are oficially supported by gatsby team? I am keen to do a pull request in this plugin to support this feature :)

**Reply by KyleAMathews** [reply]
yup, they're supported. Can't you add multiple instances of the plugin though? One for each RSS feed you want?

**Reply by lucashfreitas** [reply]
Yeap I could do that but the instances are set up in the backend, so I need to fetch them beforehand. The problem is that the plugin defines a top level array for feeds and does not allow to build them based on the data, we would have to modify `onBuild` event to fetch the data and them create N+1 RSS feeds from there. 

I ended up creating a node scripts that fetches data and them creates a json file. Then `gatsby-node `reads that JSON file and creates the configuration object that will be used by the plugin. 

Would be interesting if we could do that using gatsby graphql query ecosystem instead of fetching data using external clients as `contentful-sdk-client`, etc.

**Reply by lucashfreitas** [reply]
I will flag this as answered but it could be use as a reference if someone come across the same issue.
